02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,404 @@
+  dplyr::distinct()
+```
+
+Now let's join the mapped data from `human_zebrafish_key` with the expression data in `df`.
+
+```{r}
+human_zebrafish_mapped_df <- df %>%
+  dplyr::left_join(human_zebrafish_key, by = c("Gene" = "zebrafish_ensembl")) %>%
+  # Sort so that the columns with the zebrafish Ensembl IDs and human gene symbols are first
+  dplyr::select(Gene, human_symbol, dplyr::everything())
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_zebrafish_mapped_df, n = 25)
+```
+
+Let's get a summary of the human gene symbols returned in our mapped data frame, `human_zebrafish_mapped_df`. 
+
+```{r}
+# We can use this `count()` function after `group_by()`to get a count of how many `zebrafish_ensembl` IDs there are per `human_symbol`s
+human_zebrafish_mapped_df %>%
+  dplyr::group_by(human_symbol) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human gene symbol with the most mapped zebrafish Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped symbols!
+
+Now, let's get an idea of how many zebrafish Ensembl IDs we have that were not mapped to human gene symbols.
+
+```{r}
+sum(is.na(human_zebrafish_mapped_df$human_symbol))
+```
+
+We have 464 NAs, which means we have 464 zebrafish Ensembl IDs that were not mapped to human gene symbols.
+This is okay because we do not expect everything to map across species.
+
+## Take a look at some multi-mappings
+
+If a zebrafish Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+Let's look at the `ENSDARG00000069142` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(Gene == "ENSDARG00000069142")
+```
+
+Remember that our original data is associated to zebrafish Ensembl IDs, so when a human gene symbol maps to multiple zebrafish Ensembl IDs, the data will not get duplicated, but you will have multiple rows associated with that human symbol.
+Let's look at the `MATR3` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(human_symbol == "MATR3")
+```
+
+## Collapse multi-mapped human gene symbols
+
+Remember that if a zebrafish Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+We can collapse the multi-mapped values into a list for each Ensembl ID as to not have duplicate expression values in our data frame.
+
+In the next chunk, we show how we can collapse all the human gene symbols into one column where we store them all for each unique zebrafish Ensembl ID.
+
+```{r}
+collapsed_human_symbol_df <- human_zebrafish_mapped_df %>%
+  # Group by zebrafish Ensembl IDs
+  dplyr::group_by(Gene) %>%
+  # Collapse the mapped values in `human_zebrafish_mapped_df` into one column named `all_human_symbols`
+  dplyr::summarize(all_human_symbols = paste(human_symbol, collapse = ";")) %>%
+  # Now we need get the gene expression data back
+  dplyr::left_join(df, by = "Gene")
+
+head(collapsed_human_symbol_df)
```

Comment:
We lose the `support` column here. That may not be a problem, but it should probably be noted.